simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/station/component/chargingstation/IdentityVariableAccessorTest.java

```diff
@@ -0,0 +1,94 @@
+        return Stream.of(
+                arguments(ChargingStationComponent.NAME, IdentityVariableAccessor.NAME, SetVariableDatum.AttributeType.ACTUAL, STATION_IDENTITY_CODE, SetVariableResult.AttributeStatus.REJECTED)
+        );
+    }
+
+    static Stream<Arguments> getVariableDatumProvider() {
+        return Stream.of(
+                arguments(ChargingStationComponent.NAME, IdentityVariableAccessor.NAME, GetVariableDatum.AttributeType.ACTUAL, GetVariableResult.AttributeStatus.ACCEPTED, STATION_IDENTITY_CODE),
+                arguments(ChargingStationComponent.NAME, IdentityVariableAccessor.NAME, GetVariableDatum.AttributeType.MAX_SET, GetVariableResult.AttributeStatus.NOT_SUPPORTED_ATTRIBUTE_TYPE, null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setVariableDatumProvider")
+    void shouldValidateSetVariableDatum(String componentName, String variableName, SetVariableDatum.AttributeType attributeType, String value, SetVariableResult.AttributeStatus expectedAttributeStatus) {
+        //when
+        SetVariableResult result = variableAccessor.validate(
+                new Component().withName(new CiString.CiString50(componentName)),
+                new Variable().withName(new CiString.CiString50(variableName)),
+                attributeType,
+                new CiString.CiString1000(value)
+        );
+
+        //then
+        assertCiString(result.getComponent().getName()).isEqualTo(componentName);
+        assertCiString(result.getVariable().getName()).isEqualTo(variableName);
+        assertThat(result.getAttributeType()).isEqualTo(SetVariableResult.AttributeType.fromValue(attributeType.value()));
+        assertThat(result.getAttributeStatus()).isEqualTo(expectedAttributeStatus);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getVariableDatumProvider")
+    void shouldGetVariableDatum(String componentName, String variableName, GetVariableDatum.AttributeType attributeType, GetVariableResult.AttributeStatus expectedAttributeStatus, String expectedValue) {
+        //given
+        initStationMockIdentityCode(expectedValue);
+
+        //when
+        GetVariableResult result = variableAccessor.get(
+                new Component().withName(new CiString.CiString50(componentName)),
+                new Variable().withName(new CiString.CiString50(variableName)),
+                attributeType);
+
+        //then
+        assertCiString(result.getComponent().getName()).isEqualTo(componentName);
+        assertCiString(result.getVariable().getName()).isEqualTo(variableName);
+        assertThat(result.getAttributeType()).isEqualTo(GetVariableResult.AttributeType.fromValue(attributeType.value()));
+        assertThat(result.getAttributeStatus()).isEqualTo(expectedAttributeStatus);
+        assertCiString(result.getAttributeValue()).isEqualTo(expectedValue);
+    }
+
+    private void initStationMockIdentityCode(String expectedValue) {
+
+        if (expectedValue != null) {
```

Comment:
Please dont create tests with conditional-logic - its anti-pattern. It makes tests more complicated

Reply:
Since test is parametrized - some of the cases might not lead to calling a mock, therefore mockito complains about unnecessary stubbings. I mark mocks as lenient then and remove conditions.